### DISCUSSION TITLE
Set camera imported from glb file as view.camera

### DISCUSSION THREAD

**Discussion by TiduTidu** [google/filament#3713] [Q&A] [answered]
`scene.addEntities(asset.cameraEntities)`
with this line the glb camera is being added to the scene but I want to use the camera as my view camera
`view.camera = //glbCamera`

What I basically want to do is, create a scene in Blender with the lights and camera how I want them and then import it to android so that the render on Blender and the render on android are the same without any tweeking in the code. 

I also have some lighting problems but I can start a different discussion for that if needed

**Comment by prideout** [comment, marked as answer]
Yes it's possible to do what you need.  I think something like this would work:

```
view.camera = engine.getCameraComponent(asset.cameraEntities[0])
```

**Reply by TiduTidu** [reply]
- Rendered in Blender Eevee
- Camera Position 0,0,0 pointing at y+ ; Nearplane 0.1m Farplane 1000m
- Point light front bottom left corner 20W
![BlenderRender](https://user-images.githubusercontent.com/79167427/113273797-4bad2c00-92dd-11eb-9106-6f70d0fb5788.png)

- using filament camera; Nearplane 0.1(meters?) Farplane 1000(m); 
-  camera.projection.perspective
![RenderedFilament_nearPlane_0_1](https://user-images.githubusercontent.com/79167427/113275370-02f67280-92df-11eb-8773-a91eb10934f9.png)

- using filament camera; Nearplane 3.8(meters?) Farplane 1000(m); 
-  camera.projection.perspective
![Rendered_Filament_nearPlane_3_8](https://user-images.githubusercontent.com/79167427/113275398-0c7fda80-92df-11eb-8140-f4b2d4127e7b.png)

- Using Blender camera , view.camera = engine.getCameraComponent(asset.cameraEntities[0])
![RenderedFilament_BlenderCamera](https://user-images.githubusercontent.com/79167427/113275407-11dd2500-92df-11eb-9fc0-d31d956dfcca.png)

I didn't change the camera in Blender so that width and height are similar to that of a phone screen I guess that might be the problem for the squishing I can't tell if there are any fov problems but doesnt seem like it. 
I don't get why the near plane on the filament camera has to be brought from 0.1 to 3.8 (in this case) to view the scene properly. 
And to the lighting Issues im gonna open up another discussion soon, for example the point light is at 20W visible in Blender but not in Filament if I juice it up alot the Blender render looks horrible but the light can be noticed in Filament, 
maybe because Filament uses lumen or something idk.

Ty for the help

**Reply by romainguy** [reply]
20W is a dim light, you'll have to change the default camera exposure which is set for bright daylight outdoors.

**Reply by bejado** [reply]
See the comments on the `Camera.setScaling` method regarding the aspect ratio. AFAIK Blender always sets the aspect ratio to 1 when exporting a glTF file.

**Reply by TiduTidu** [reply]
![Exposure](https://user-images.githubusercontent.com/79167427/113342434-c5213a80-932e-11eb-896e-5932c3abf8c7.png)

Looks like I found the exposure gonna try some stuff out also with the scaling. Thank you for the help guys

**Reply by romainguy** [reply]
Another interesting thing to note: Filament always uses a far plane set to infinity. The far plane you specify is used for culling and lighting, so you should not set it so far if possible (esp. for a scene like this :).

**Reply by TiduTidu** [reply]
I appreciate every bit of info ty. 
If there are no objects in the far that can interact with the light, would the performance still suffer from that? and if I use a camera thats importet from an glb, blender in that instanse would use the farplane as a max render distance and filament would use the same farplane for lighting and culling is that right?
